### PR TITLE
Added CodeClimate quality and coverage badges

### LINKED ISSUE #178: Integration with Code Climate?
### Scenario/Intent
Code Climate is a pretty cool and known cloud based tool to analyze our code quality, and since we have ruby code in our providers it would be a good to integrate it here.

The current oneview-chef status can be seen [here](https://codeclimate.com/github/HewlettPackard/oneview-chef).

Any oppositions, thoughts or suggestions?

### PR DESCRIPTION
**Pull request by tmiotto**
### Description
Adds CodeClimate quality and coverage badges for the repo.

:warning:  Coverage only shows the coverage for the master branch, also it will only show the report generated by the simplecov (So no ChefSpec coverage)

### Issues Resolved
#178 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [ ] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,3 +8,4 @@ before_install: gem update bundler
 
 script:
   - rake test
+  - codeclimate-test-reporter
```
Comment (jsmartt) [inline .travis.yml]: The [docs](https://docs.codeclimate.com/v1.0/docs/travis-ci-ruby-test-coverage) mention to prepend this line with `bundle exec`, but it appears to be working for now, so I guess it's OK

**Review by jsmartt — APPROVED** [review]
Looks good to me!

**Comment by fgbulsoni** [comment]
Only for master? It'd probably be better to have it run against any/all branches, so we have access to code coverage before merging.
Is this something coming from codeclimate itself or from coveralls.io? If it´s from coveralls, probably best to just go straight to them instead of the middle-man.

To have this working directly with coveralls:
1. add `require 'coveralls'` to the top of the `spec_helper.rb` file
2. add the following to the `spec_helper.rb` file before all the simplecov defines:
```ruby
Coveralls.wear!

SimpleCov.formatters = [
  SimpleCov::Formatter::HTMLFormatter,
  Coveralls::SimpleCov::Formatter
]
```
3. add `gem 'coveralls'` to the gemfile
4. add to readme `
[Coverage Status](https://coveralls.io/github/HewlettPackard/oneview-sdk-ruby?branch=master)`

This should get coveralls to work on all branches and add an extra status check besides the PR/Push from travis.

**Comment by tmiotto** [comment]
@fgbulsoni But it is not using the Coveralls, just the CodeClimate

**Review by fgbulsoni — APPROVED** [review]

**Comment by fgbulsoni** [comment]
okay then :shipit:

File: .travis.yml (hunk 1 above)
Comment (tmiotto) [reply .travis.yml]: It won't matter because of our build steps on Travis. But it won't hurt to add the `bundler exec`

**Pull request merged**